### ISSUE TITLE
GRPO early termination

### ISSUE THREAD

**Issue by OgasTm** [modelscope/ms-swift#5075]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

I'm performing a primitive test with GRPO on my custom dataset.

Format of my dataset:

`{"messages": [{"role": "user", "content": "<video>Please identify the gender of the patient in the video. Please answer with \"female\" or \"male\". Do not include extra text in your output—only the answer."}], "videos": ["/mnt/..._v2.mp4"], "solution": "male"}`

All video paths are valid in my dataset.

```
#!/usr/bin/env bash
export PYTHONUNBUFFERED=1
export OMP_NUM_THREADS=1
export MKL_NUM_THREADS=1
export TOKENIZERS_PARALLELISM=false
export CUDA_VISIBLE_DEVICES=1
export PYTORCH_DISABLE_WEIGHTS_ONLY_LOAD=1
export TRANSFORMERS_VERBOSITY=debug
export SWIFT_LOG_LEVEL=DEBUG
export MAX_PIXELS=$((512*512))
export MIN_PIXELS=$((256*256))
export FPS=1
export FPS_MIN_FRAMES=2
export FPS_MAX_FRAMES=4
# export PYTORCH_CUDA_ALLOC_CONF=max_split_size_mb:128,expandable_segments:True
export TORCH_USE_CUDA_DSA=0

swift rlhf \
  --rlhf_type grpo \
  --model Qwen/Qwen2.5-VL-3B-Instruct \
  --dataset /mnt/.../grpo_train_filtered.jsonl \
  --reward_funcs accuracy \
  --train_type lora \
  --lora_rank 8 \
  --lora_alpha 16 \
  --torch_dtype bfloat16 \
  --max_length 2048 \
  --max_completion_length 256 \
  --beta 0.02 \
  --epsilon_high 0.2 \
  --num_train_epochs 1 \
  --per_device_train_batch_size 1 \
  --per_device_eval_batch_size 2 \
  --learning_rate 1e-6 \
  --gradient_accumulation_steps 2 \
  --eval_steps 100 \
  --save_steps 100 \
  --save_total_limit 3 \
  --logging_steps 5 \
  --output_dir /mnt/SSD3/detao/grpo_output_test \
  --warmup_ratio 0.05 \
  --dataloader_num_workers 0 \
  --num_generations 2 \
  --temperature 1.0 \
  --log_completions true \
  --logging_first_step true \
  --ignore_data_skip true
```

It terminates after 2 steps, regardless of specific values of-- num_generations,-- gradient_accumulation_steps, 

'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 6e-08, 'memory(GiB)': 22.82, 'train_speed(iter/s)': 0.013871, 'completions/mean_length': 2.0, 'completions/min_length': 2.0, 'completions/max_length': 2.0, 'completions/clipped_ratio': 1.0, 'rewards/MathAccuracy/mean': 0.0, 'rewards/MathAccuracy/std': 0.0, 'reward': 0.0, 'reward_std': 0.0, 'kl': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.0, 'global_step/max_steps': '1/358', 'percentage': '0.28%', 'elapsed_time': '1m 11s', 'remaining_time': '7h 5m 43s'}
Train:   1%|▋                                                                                                                     | 2/358 [02:05<6:02:05, 61.03s/it] ... $ /usr/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
ms swift 3.6.0
torch 2.7.0
CUDA 12.4
4 * A6000

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by hjh0119** [comment]
can you retry with any open-source dataset that I can reproduce?

**Comment by OgasTm** [comment]
It appears to be a problem with my dataset video. I changed the videos to ones from the Motion Bench dataset, and the steps didn't stop at step 2. I will investigate further.
